src/Jobs/StructureOrders.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace RecursiveTree\Seat\TreeLib\Jobs;
+
+use Illuminate\Support\Facades\DB;
+use RecursiveTree\Seat\TreeLib\Models\MarketOrder;
+use Seat\Eseye\Exceptions\RequestFailedException;
+use Seat\Eveapi\Jobs\AbstractAuthCharacterJob;
+use Seat\Eveapi\Jobs\EsiBase;
+use Seat\Eveapi\Models\RefreshToken;
+use Seat\Eveapi\Models\Universe\UniverseStructure;
+
+/**
+ * Class Orders.
+ *
+ * @package Seat\Eveapi\Jobs\Market
+ */
+class StructureOrders extends AbstractAuthCharacterJob
+{
+
+    /**
+     * @var string
+     */
+    protected $method = 'get';
+
+    /**
+     * @var string
+     */
+    protected $endpoint = '/markets/structures/{structure_id}/';
+
+    /**
+     * @var string
+     */
+    protected $version = 'v1';
+
+    /**
+     * @var string
+     */
+    protected $scope = 'esi-markets.structure_markets.v1';
+
+    /**
+     * @var array
+     */
+    protected $tags = ['market', 'structure'];
+
+    /**
+     * @var int The structure ID to which this job is related.
+     */
+    protected $structure_id;
+
+    public function __construct(RefreshToken $token, UniverseStructure $structure)
+    {
+        $this->structure_id = $structure->structure_id;
+
+        parent::__construct($token);
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @throws \Throwable
+     */
+    public function handle()
+    {
+        $structure_id = $this->getStructureId();
+        $structure = UniverseStructure::find($structure_id);
+
+        MarketOrder::where('location_id', '=', $this->getStructureId())
```

Comment:
I see you have another delete at the end. Is there a reason for having two?

Reply:
A better way would be to add a "updated_at" colum to Orders Table.
Then we can delete all record that have not been update since the start of the job (it mean they don't exist anymore).

With current implementation we get an empty table, and if a job relying on that data gets executed in another thread at the same time, (or worse is observing market_orders update) it's going have false data.

One "not sexy" way of doing it to get a list of all upserted "order_id" and remove all orders that are not in that list.
But that would be a big "WHERE NOT IN (...)" request which on some platform fail with too much data.
Or we would need to parse all market_orders table for that structure one by one to find which one does not exist anymore in the market.

Please ask for that column to be added into seat so we handle that better in a future update.

Reply:
I agree, I'll try to push it